src/com/amazon/ion/impl/lite/IonValueLite.java

```diff
@@ -979,10 +990,21 @@ protected int hashTypeAnnotations(final int original)
             return original;
         }
 
-        int result = valueHashSalt * original + tokens.length;
+        // Note: Because the tokens array doubles in size when it grows, tokens.length does not correctly convey
+        // the actual number of annotations on the value.
+        int count = 0;
+        for (SymbolToken token : tokens) {
+            if (token == null) {
+                break;
+            }
+            count++;
+        }
```

Comment:
```suggestion
        int count = 0;
        while (tokens[count] != null && ++count < tokens.length);
```

Because Java does not have `until`, or a built-in findFirst or indexOf for arrays :(

Or if the above is less clear:
```java
        int count = 0;
        while (count < tokens.length && tokens[count] != null) {
            count++;
        }
```

This is all 100% optional, I mostly just played around with this because Java's lack of ability to express this more succinctly was irritating.